tests/Propel/Tests/Generator/Builder/Om/GeneratedObjectM2MRelationPolymorphicWithSelfReference.php

```diff
@@ -0,0 +1,133 @@
+<?php namespace Propel\Tests\Generator\Builder\Om;
+
+use Propel\Generator\Platform\MysqlPlatform;
+use Propel\Generator\Util\QuickBuilder;
+use Propel\Runtime\Collection\ObjectCollection;
+use Propel\Runtime\Collection\ObjectCombinationCollection;
+use Propel\Tests\Helpers\PlatformDatabaseBuildTimeBase;
+
+
+class GeneratedObjectM2MRelationPolymorphic extends PlatformDatabaseBuildTimeBase
+{
+    /**
+     * Tests for a M2M relation with three pks where each is a FK.
+     *
+     * @group database
+     */
+    protected $databaseName = 'migration';
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (!class_exists('\RelationpmsrUserQuery')) {
+            $schema = '
+    <database name="migration" schema="migration">
+        <table name="relationpmsr_user_child" isCrossRef="true">
+            <column name="user_id" type="integer" primaryKey="true"/>
+            <column name="child_type" primaryKey="true"/>
+            <column name="child_id" type="integer" primaryKey="true"/>       
+
+            <foreign-key foreignTable="relationpmsr_user" phpName="User" onDelete="cascade">
+                <reference local="user_id" foreign="id"/>
+            </foreign-key>            
+
+            <foreign-key foreignTable="relationpmsr_position" phpName="Position" onDelete="cascade">
+                <reference local="child_id" foreign="id"/>
+                <reference local="child_type" value="position"/>
+            </foreign-key>                        
+            
+            <foreign-key foreignTable="relationpmsr_user" phpName="ChildUser" onDelete="cascade">
+                <reference local="child_id" foreign="id"/>
+                <reference local="child_type" value="user"/>
+            </foreign-key>
+        </table>
+
+        <table name="relationpmsr_user">
+            <column name="id" type="integer" primaryKey="true" autoIncrement="true"/>
+            <column name="name" />
+        </table>
+
+        <table name="relationpmsr_position">
+            <column name="id" type="integer" primaryKey="true" autoIncrement="true"/>
+            <column name="name" />
+        </table>                
+    </database>
+        ';
+
+            $this->buildAndMigrate($schema);
+        }
+    }
+
+    /**
+     *
+     *           add     |    remove     |    set     |    get
+     *        +---------------------------------------------------
+     * add    |    1             2             3            4
+     * remove |    5             6             7            8
+     * set    |    9            10            11           12
+     *
+     */
+
+
+
+    /*
+     * ####################################
+     * 1. add, add
+     */
+    public function test1()
```

Comment:
```suggestion
    public function testCombinations()
```

I suspect this test should be broke into three tests, based on the cyclical appearance of setup + assert code in here.